imap/mailbox.c

```diff
@@ -962,6 +962,7 @@ static int mailbox_open_index(struct mailbox *mailbox)
 static int mailbox_mboxlock_reopen(struct mailboxlist *listitem, int locktype, int index_locktype)
 {
     struct mailbox *mailbox = &listitem->m;
+    uint32_t legacy_dirs = (mailbox_mbtype(mailbox) & MBTYPE_LEGACY_DIRS);
```

Comment:
It's a little jarring to see a variable that's only used as a boolean being declared as an unsigned type with a fixed-size.  I guess this is because mbtype itself is uint32_t, and the compiler complains if the type doesn't match?

If we wanted to use a normal int instead, I think we could do something like:

```
    int legacy_dirs = !!(mailbox_mbtype(mailbox) & MBTYPE_LEGACY_DIRS);
```

Looks like we already use the `!!` construct for this sort of thing in a few other places, so it's not without precedent.

Reply:
yeah, we could - there was already one that was this, so I just copied it